src/structures/BaseManager.ts

```diff
@@ -1,8 +1,21 @@
+import { FetchOptions, PaginationMeta } from "../common";
+import { ClientMeta } from "../common/client";
+import { Dict } from "./Dict";
+import { ValidationError } from "./Errors";
+
+function isMetadata(meta: PaginationMeta | ClientMeta): meta is PaginationMeta {
+    return (meta as PaginationMeta).totalPages !== undefined;
+}
+
 export abstract class BaseManager {
+    declare meta: PaginationMeta | ClientMeta | undefined;
```

Comment:
No need for `declare` here, also is `undefined` necessary?

Reply:
I think `declare` makes more sense here, as it's not defined here, but sure. Undefined is definitely necessary.